Feature/project docs (#367)
redballoonsecurity/ofrak: docs/user-guide/gui/projects.md

# Projects
Projects are OFRAK's way of saving, organizing, and sharing your work. Projects are collections of binaries and scripts along with metadata that describes relationships between them.

## The Interface
The Project Options interface is available from the start menu and gives you three options for opening a project. The "Create New Project" opens up a blank project, "Open Existing Project" will open a project already in your Project Path, and "Clone Project from Git" clones the URL provided to your Project Path. The Project Path, where all of the projects are saved to and loaded from, may be modified in the Advanced Options section of the Project Options interface.

![](assets/project-options.png)

> The OFRAK Project Manager interface is loaded upon opening a project and displays the structure of the project along with options to modify it. The Binaries panel displays each binary in the project. Select a binary to work with by clicking on its name. The Script panel shows options for each script in your project. Upon selecting a binary, two checkboxes options appear for each script. Hovering over each checkbox describes its function. One check box selects whether the script is compatible with the currently selected binary, allowing access to it in the Run Script option of the OFRAK Resource toolbar. The second checkbox selects a script to be run upon entering the OFRAK Resource interface, giving a starting point for exploration.

Reviewer: ```suggestion
The OFRAK Project Manager interface is loaded upon opening a project. It displays the structure of the project along with options to modify it. The Binaries panel displays each binary in the project. Select a binary to work with by clicking on its name. The Script panel shows options for each script in your project. Upon selecting a binary, two checkboxes options appear for each script. Hovering over each checkbox describes its function. One check box selects whether the script will run with the currently selected binary, allowing access to it in the Run Script option of the OFRAK Resource toolbar. The second checkbox selects a script to be run upon entering the OFRAK Resource interface, giving a starting point for exploration.
```

Author: changed to "can be run on" because i thin "will" makes it sound like it will run on startup

![](assets/project-manager.png)

## Creating a Project

> OFRAK projects should be initialized through the OFRAK Project Manager to ensure a correcting project structure. Binaries and scripts are added to the project thorugh the corresponding toolbar options on the left side of the interface. After adding your assets, use the checkboxes in the script panel to link scripts with their binaries. The Save option in the toolbar will save your project, and reset will go back to the previously saved state.

Reviewer 2: ```suggestion
OFRAK projects should be initialized through the OFRAK Project Manager to ensure a correct project structure. Binaries and scripts are added to the project through the corresponding toolbar options on the left side of the interface. After adding your assets, use the checkboxes in the script panel to link scripts with their binaries. The Save option in the toolbar will save your project, and Reset will go back to the previously saved state.
```

## Launching a Project Binary

> Binaries from projects can be launched from the Launch option of the toolbar. When launched, the selected script will run on the binary and then display the result in the OFRAK Resource interface.

Reviewer 2: ```suggestion
Binaries from projects can be launched from the Launch option of the toolbar. When launched, the selected binary's init script (marked by the second checkbox) will run on the binary and then display the result in the OFRAK Resource interface.
```
